### ISSUE TITLE
3.2.7: Controls that require scrolling to reach

### ISSUE THREAD

**Issue by benja11y** [w3c/wcag#1875]
We gather that controls that the user can scroll to would pass this SC. We gather the following scenarios would pass:

* T&C modals that require a user to scroll to get to the controls that allow them to accept the terms.
* Cases where more options/fields become visible because of a selection elsewhere on the form. Think of filling out an address and the City field is not visible until the State is selected.
* Similarly, an e-reader that allows the user to select some text and then pops up some buttons to allow the user to interact in different ways with that text would pass.

If that's the case, it may be worth clarifying that scrolling is acceptable.

However, what would happen in the following two different cases?

* With the examples above _if there are no visible scrollbars_ to alert the user to the fact that there's more content to scroll through before they get to the controls required to proceed?
* If table column sort controls only appear on hover/focus, but the user needs to use them in order to find the information they're looking for in the table?

This issue has been raised on behalf of [TPGi](https://www.tpgi.com), based on an internal consultation process. [An article containing all of our comments](https://www.tpgi.com/wcag-22-working-draft-feedback/) can be found on our blog, along with an introduction and acknowledgments.

**Comment by wardav** [comment]
This is one that came up from my team, as well. A common pattern is that you have to scroll a window (filled with T&Cs) for a checkbox (consenting to the T&Cs) to appear.

As with yourself, my understanding is that these **would** pass, but that it would be good to make this crystal clear. For example:

> Controls that become visible due to some mechanism other than mouse hover or keyboard focus are not in scope. 
> 
> For example, if you are required to first complete form fields or scroll a window to prompt controls to appear, the requirements for this would be Keyboard, Focus Order, ...  
> 
> That said, the success criterion would be in scope if any of the steps involved hidden controls that needed mouse hover or keyboard focus. For example, if the scrollbar on a modal dialog must be scrolled to the bottom to prompt a checkbox to appear, that scrollbar must not require keyboard focus or mouse hover to be visible. 

The only thing I am wondering is what is the behaviour is driven by the user agent? What if some browsers have scroll bars that only appears when mouse hover or keyboard focus is in the window?

**Comment by alastc** [comment]
I'm not sure I'm understanding the issue, the trigger for the SC scope is hover & focus, I don't see how scrolling fits into this?

My initial thoughts:

> T&C modals that require a user to scroll to get to the controls that allow them to accept the terms.

Unless the 'accept' button only appears on hover/focus of something, then it isn't in scope.

> Cases where more options/fields become visible because of a selection elsewhere on the form. Think of filling out an address and the City field is not visible until the State is selected.

Unless triggered by hover/focus, not in scope.

> Similarly, an e-reader that allows the user to select some text and then pops up some buttons to allow the user to interact in different ways with that text would pass.

Selection isn't hover/focus, so not in scope.

> If table column sort controls only appear on hover/focus, but the user needs to use them in order to find the information they're looking for in the table?

That does appear to be in scope, therefore you'd need an indicator that there are sort controls. I.e. more than just the text in the TH, such as a up/down arrow.

**Comment by DavidMacDonald** [comment]
> T&C modals that require a user to scroll to get to the controls that allow them to accept the terms.

For the scrolling question, I could see that a submit button at the end of a long privacy scrollable section could become visible on focus if the user tabs through the scroll window and it auto scrolls into view when the user focuses on it with tab key. That would be in scope. We weren't thinking about that scenario (at least I wasn't) but the presence of a scrollbar would pass it.

Agree with other responses by Alastair

 If there is no scroll bar then, yes, I would say it fails because there is no indicator that there is interactive content. I think it requires discussion by the group.

**Comment by alastc** [comment]
Thanks @DavidMacDonald.

FYI, it helps to keep each issue's changes in a separate branch. I've separated that out into a separate branch and done some edits in #2044
